manifests/foreman_proxy_content.pp

```diff
@@ -24,7 +24,7 @@
   validate_present($foreman_proxy_fqdn)
 
   if $foreman_proxy_fqdn == $::fqdn {
-    warning('The hostname is the same as the foreman-proxy')
+    fail('The hostname is the same as the foreman-proxy')
```

Comment:
I'd consider one slight modification to the wording -- `The hostname is the same as the provided hostname for the foreman-proxy`